rules/regex-shorthand.js

```diff
@@ -11,6 +11,10 @@ const create = context => {
 
 			const newPattern = cleanRegexp(oldPattern, flags);
 
+			if (node.parent.type === 'NewExpression' && node.parent.callee.name === 'RegExp') {
```

Comment:
Can you add a short comment about what this is for. Something like `// Handle regex literal inside RegExp constructor in the other handler`